File: problem_8/problem_8.js
const now = require("performance-now")
fs = require("fs");

const problem8 = () => {
    fs.readFile("./problem_8.txt", "utf8", (err, data) => {
        if (err){ throw err; }
        const start = now();
        const number = [];
        for (const char of data){ 
            if (char !== "\n"){
                number.push(parseInt(char, 10))
            }
        }
        let max_product = 0;
        for (var i = 0; i < number.length - 13; i ++){
            let product = number.slice(i,i+13).reduce((a,b)=>a*b,1)
            if (product > max_product){ 
                max_product = product;
            }
        }
        console.log(`${max_product} found in ${(now() - start) / 1000} seconds.`);
    })
}

problem8()
// 23514624000 found in 0.0029508629999999984 seconds.

File: problem_8/problem_8.py
# The four adjacent digits in the 1000-digit number that have the greatest product are 9 × 9 × 8 × 9 = 5832.
# Find the thirteen adjacent digits in the 1000-digit number that have the greatest product. What is the value of this product?

# We are given a 20 by 50 (height by width) grid


import time
from functools import reduce
from queue import LifoQueue, Queue

def problem_8():
    number = [int(n) for n in open("./problem_8.txt", "r").read() if n != "\n"]
    max_product = 0

    for i in range(len(number) - 13):
        product = reduce( (lambda a,b:a*b), number[i:i+13])
        if product > max_product:
            max_product = product
            
    return max_product

start = time.time()
solution = problem_8()
print(f"{solution} found in {time.time() - start} seconds.")
# 23514624000 found in 0.003715038299560547 seconds.
